### PR TITLE
Restore a noop _all metadata field for 6x indices

### LINKED ISSUE #37429: Elasticsearch node on 7.0 alpha is not booting (Upgrade from 6.1.2)
Elasticsearch version (bin/elasticsearch --version): 7.0.0 alpha1

Plugins installed: [analysis-phonetic, ingest-attachment]

JVM version (java -version): 1.8.0.181

OS version (uname -a if on a Unix-like system): Windows 2012 R2

Description of the problem including expected versus actual behavior: We have disabled "_all" field in our mappings in ElasticSearch 6.1.2. We have created indices and indexed data into it. The following is an example of the mapping.

{
  "testindex": {
    "mappings": {
      "type": {
        "_all": {
          "enabled": false
        },
        "properties": {
          "title": {
            "type": "text"
          },
          "description": {
            "type": "text"
          }
        }
      }
    }
  }
}

Everything is fine with ElasticSearch 6.1.2. But when we upgrade to Elasticsearch 7.0.0 alpha 1, ElasticSearch boot up failed with the following error.

org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters:  [_all : {enabled=false}]

Is there a way to ignore it (_all) while booting up?
Or is there a way to update the mapping to remove the _all field?

Looking for a solution other than mandating re-index of data.

Please provide your suggestions. Thanks.

### PR DESCRIPTION
**Pull request by jimczi**
This commit restores a noop version of the AllFieldMapper that is instanciated only
for indices created in 6x. We need this metadata field mapper to be present in this version
in order to allow the upgrade of indices that explicitly disable _all (enabled: false).
The mapping of these indices contains a reference to the _all field that we cannot remove
in 7 so we'll need to keep this metadata mapper in 7x. Since indices created in 6x will not
be compatible with 8, we'll remove this noop mapper in the next major version.

Closes #37429

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/indices/IndicesModuleTests.java [hunk 1]
```diff
@@ -87,26 +90,50 @@ public MetadataFieldMapper getDefault(MappedFieldType fieldType, ParserContext c
...
+        {
+            Version version = VersionUtils.randomVersionBetween(random(), Version.V_6_0_0, Version.V_7_0_0.minimumCompatibilityVersion());
+            assertFalse(module.getMapperRegistry().getMapperParsers().isEmpty());
+            assertFalse(module.getMapperRegistry().getMetadataMapperParsers(version).isEmpty());
+            Map<String, MetadataFieldMapper.TypeParser> metadataMapperParsers =
+                module.getMapperRegistry().getMetadataMapperParsers(version);
+            int i = 0;
+            for (String field : metadataMapperParsers.keySet()) {
+                assertEquals(EXPECTED_METADATA_FIELDS_6x[i++], field);
+            }
```
Comment (jpountz) [inline server/src/test/java/org/elasticsearch/indices/IndicesModuleTests.java]: do we need to check that the have the same size as well?

**Review by jpountz — APPROVED** [review]
Thanks @jimczi.

File: qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/10_basic.yml [hunk 2]
```diff
@@ -125,3 +125,9 @@
         wait_for_completion: true
         task_id: $task_id
   - match: { task.headers.X-Opaque-Id: "Reindexing Again" }
+
+---
+"Index with _all is available":
+  - do:
+      indices.get:
+        index: all-index
```
[the remark this answers is not in the document]
Comment (jimczi) [reply qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/10_basic.yml]: https://github.com/elastic/elasticsearch/pull/37808/commits/cdaa804e87ec5a7a276d31e6a71eb06d8d69924c

File: server/src/test/java/org/elasticsearch/indices/IndicesModuleTests.java (hunk 1 above)
Comment (jimczi) [reply server/src/test/java/org/elasticsearch/indices/IndicesModuleTests.java]: https://github.com/elastic/elasticsearch/pull/37808/commits/cdaa804e87ec5a7a276d31e6a71eb06d8d69924c

**Comment by jimczi** [comment]
Jenkins run elasticsearch-ci/1

**Comment by jimczi** [comment]
Jenkins run elasticsearch-ci/2

**Comment by jimczi** [comment]
Jenkins run elasticsearch-ci/2

**Comment by jimczi** [comment]
Jenkins run elasticsearch-ci/2

**Pull request merged**